File: 0704-binary-search/0704-binary-search.cpp
class Solution {
public:
    int search(vector<int>& nums, int target) {
        int n=nums.size();   
        int low=0; 
        int high=n-1;

        while(low<=high){
            // Till the Time low is less than or equal to high this will be true 
            int guess=(low+high)/2;           // Random Element is taken as the Mid one 
           
            if(nums[guess]==target){
                // If the Element guessed is found 
                return guess;
            }

            if(nums[guess]>target){
                // the greater one 
                high=guess-1;
            }
            else{
                // the smaller or lesser one 
                low=guess+1;
            }
        }
        return -1;                     // If the element is not present at all 
    }
};

File: 0704-binary-search/0704-binary-search.go
func search(nums []int, target int) int {
    n:=len(nums)
    low:=0
    high:=n-1

    for low<=high{
        guess:=(low+high)/2

        if nums[guess]==target{
            return guess
        }

        if nums[guess]<target{
            low=guess+1
        }else{
            high=guess-1
        }
    }
    return -1
}